### ISSUE TITLE
Unexpected false-positive "no-value-for-parameter" from type stubs 'hidden' under 'if TYPE_CHECKING:'

### ISSUE THREAD

**Issue by geokala** [pylint-dev/pylint#8689]
### Bug description

This may not be a bug in pylint, so possibly can just be closed but as I just burned a bunch of time figuring it out, I'm raising it here.

When type stubs for a function that needs to be decorated are hidden under a TYPE_CHECKING, pylint seems to get unhappy with them- when they're in separate files, it can be a bit puzzling until one notices the issue is actually with the type stub.

Code to repro attached (as txt because github doesn't like py extensions).
[testfile.txt](https://github.com/pylint-dev/pylint/files/11477908/testfile.txt)

### Command used

```shell
pylint --signature-mutators=add_an_arg testfile.py
```

### Pylint output

```shell
************* Module testfile
testfile.py:47:8: E1120: No value for argument 'something_else' in method call (no-value-for-parameter)

------------------------------------------------------------------
Your code has been rated at 8.00/10 (previous run: 8.00/10, +0.00)
```

### Expected behavior

I was expecting no complaints, but I'm not sure if that was a reasonable expectation. It might've been nice to have more indication as to where the failing function call was (as this would've immediately pointed me at the type stub), but I'm not sure how feasible that is.

### Pylint version

```shell
pylint 2.17.4
astroid 2.15.5
Python 3.11.3 (main, Apr  5 2023, 14:15:06) [GCC 9.4.0]
```

**Comment by geokala** [comment]
Just to note: the fix is to decorate the type stub as well.
